presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -202,21 +202,22 @@ public PlanNode visitWindow(WindowNode node, RewriteContext<Void> context)
         {
             PlanNode source = context.rewrite(node.getSource());
 
-            ImmutableMap.Builder<Symbol, WindowNode.Function> functions = ImmutableMap.builder();
+            ImmutableMap.Builder<Symbol, WindowNode.FunctionWithFrame> functions = ImmutableMap.builder();
             ImmutableMap.Builder<WindowNode.Function, WindowNode.Frame> frames = ImmutableMap.builder();
-            for (Map.Entry<Symbol, WindowNode.Function> entry : node.getWindowFunctions().entrySet()) {
+            for (Map.Entry<Symbol, WindowNode.FunctionWithFrame> entry : node.getWindowFunctions().entrySet()) {
                 Symbol symbol = entry.getKey();
                 Symbol canonical = canonicalize(symbol);
 
-                FunctionCall functionCall = entry.getValue().getFunctionCall();
-                Signature signature = entry.getValue().getSignature();
-                WindowNode.Function function = new WindowNode.Function((FunctionCall) canonicalize(functionCall), signature);
-                functions.put(canonical, function);
+                FunctionCall functionCall = entry.getValue().getFunction().getFunctionCall();
+                Signature signature = entry.getValue().getFunction().getSignature();
+                WindowNode.Function canonicalFunction = new WindowNode.Function((FunctionCall) canonicalize(functionCall), signature);
 
-                WindowNode.Frame frame = node.getFrames().get(entry.getValue());
-                frames.put(function, new WindowNode.Frame(frame.getType(),
+                WindowNode.Frame frame = entry.getValue().getFrame();
+                WindowNode.Frame canonicalFrame = new WindowNode.Frame(frame.getType(),
                         frame.getStartType(), canonicalize(frame.getStartValue()),
-                        frame.getEndType(), canonicalize(frame.getEndValue())));
+                        frame.getEndType(), canonicalize(frame.getEndValue()));
+
+                functions.put(canonical, new WindowNode.FunctionWithFrame(canonicalFunction, canonicalFrame));
```

Comment:
Since `canonical` is not used outside this place, please inline to avoid unnecessary local variable.